scan.py

```diff
@@ -12,94 +13,161 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-import boto3
-import clamav
 import copy
 import json
-import metrics
-import urllib
-from common import *
-from datetime import datetime
+import os
+from urllib.parse import unquote_plus
+from distutils.util import strtobool
+
 from requests import patch, post
 from os.path import basename
 
-ENV = os.getenv("ENV", "")
+import boto3
+
+import clamav
+import metrics
+from common import AV_DEFINITION_S3_BUCKET
+from common import AV_DEFINITION_S3_PREFIX
+from common import AV_DELETE_INFECTED_FILES
+from common import AV_PROCESS_ORIGINAL_VERSION_ONLY
+from common import AV_SCAN_START_METADATA
+from common import AV_SCAN_START_SNS_ARN
+from common import AV_SIGNATURE_METADATA
+from common import AV_STATUS_CLEAN
+from common import AV_STATUS_INFECTED
+from common import AV_STATUS_METADATA
+from common import AV_STATUS_SNS_ARN
+from common import AV_STATUS_SNS_PUBLISH_CLEAN
+from common import AV_STATUS_SNS_PUBLISH_INFECTED
+from common import AV_TIMESTAMP_METADATA
+from common import SNS_ENDPOINT
+from common import S3_ENDPOINT
+from common import create_dir
+from common import get_timestamp
+
 
+def event_object(event, event_source="s3"):
 
-def event_object(event):
+    # SNS events are slightly different
+    if event_source.upper() == "SNS":
+        event = json.loads(event["Records"][0]["Sns"]["Message"])
+
+    # Break down the record
+    records = event["Records"]
+    if len(records) == 0:
+        raise Exception("No records found in event!")
+    record = records[0]
+
+    # sharpr update - unlike the original upstream, we (sharpr) get the bucket and key names from `Message`
     message = json.loads(event['Records'][0]['Sns']['Message'])
-    bucket = message['bucket']
-    key = urllib.unquote_plus(message['key'].encode('utf8'))
-    if (not bucket) or (not key):
-        print("Unable to retrieve object from event.\n%s" % event)
-        raise Exception("Unable to retrieve object from event.")
-    return s3.Object(bucket, key)
+    bucket_name = message['bucket']
+    key_name = unquote_plus(message['key'])
+
+    # sharpr update - this is how upstream was getting bucket and key
+    # s3_obj = record["s3"]
+
+    # # Get the bucket name
+    # if "bucket" not in s3_obj:
+    #     raise Exception("No bucket found in event!")
+    # bucket_name = s3_obj["bucket"].get("name", None)
+
+    # # Get the key name
+    # if "object" not in s3_obj:
+    #     raise Exception("No key found in event!")
+    # key_name = s3_obj["object"].get("key", None)
+
+    # if key_name:
+    #     key_name = unquote_plus(key_name)
```

Comment:
My changes. The upstream code gets bucket and key values differently than we do. Ours come from SNS in a JSON string. This reflects that and comments out their code. I thought it would be easier to leave their code commented out for when we merge upstream in again in the future.